### PR TITLE
[docs] Supported models tables

### PR DESCRIPTION
**Pull request by MKhalusova**
As a suggestion, this PR moves the tables listing supported models to the index page since the supported methods are also listed there. 

In addition, this PR adds links to the quicktour and conceptual guides to help beginners navigate the docs.

### CODE REVIEW

File: docs/source/index.mdx [hunk 1]
```diff
@@ -16,11 +16,95 @@ specific language governing permissions and limitations under the License.
...
+
+## Supported methods
 
 1. LoRA: [LORA: LOW-RANK ADAPTATION OF LARGE LANGUAGE MODELS](https://arxiv.org/pdf/2106.09685.pdf)
 2. Prefix Tuning: [Prefix-Tuning: Optimizing Continuous Prompts for Generation](https://aclanthology.org/2021.acl-long.353/), [P-Tuning v2: Prompt Tuning Can Be Comparable to Fine-tuning Universally Across Scales and Tasks](https://arxiv.org/pdf/2110.07602.pdf)
 3. P-Tuning: [GPT Understands, Too](https://arxiv.org/pdf/2103.10385.pdf)
 4. Prompt Tuning: [The Power of Scale for Parameter-Efficient Prompt Tuning](https://arxiv.org/pdf/2104.08691.pdf) 
+
+## Supported models
+
```
Comment (stevhliu) [inline docs/source/index.mdx:31]: Maybe we can add a brief description here like, "The tables below include supported PEFT methods and models for each task. Take a look at the Task guides to learn how to apply a specific PEFT method for a task."

**Review by stevhliu — APPROVED** [review]
Very nice! I think placing it on the index immediately gives users an idea of whether their use case is supported or not 👍

File: docs/source/index.mdx [hunk 2]
```diff
@@ -16,11 +16,101 @@ specific language governing permissions and limitations under the License.
...
-Supported methods include:
+If you are new to PEFT, get started by reading the [Quicktour](quicktour) guide and conceptual guides for [LoRA](/conceptual_guides/lora) and [Prompting](/conceptual_guides/prompting) methods. 
+
+## Supported methods
 
 1. LoRA: [LORA: LOW-RANK ADAPTATION OF LARGE LANGUAGE MODELS](https://arxiv.org/pdf/2106.09685.pdf)
 2. Prefix Tuning: [Prefix-Tuning: Optimizing Continuous Prompts for Generation](https://aclanthology.org/2021.acl-long.353/), [P-Tuning v2: Prompt Tuning Can Be Comparable to Fine-tuning Universally Across Scales and Tasks](https://arxiv.org/pdf/2110.07602.pdf)
 3. P-Tuning: [GPT Understands, Too](https://arxiv.org/pdf/2103.10385.pdf)
 4. Prompt Tuning: [The Power of Scale for Parameter-Efficient Prompt Tuning](https://arxiv.org/pdf/2104.08691.pdf) 
+
```
Comment (pacman100) [inline docs/source/index.mdx:29]:
```suggestion
5. AdaLoRA: [Adaptive Budget Allocation for Parameter-Efficient Fine-Tuning](https://arxiv.org/abs/2303.10512) 
6. [LLaMA-Adapter: Efficient Fine-tuning of Language Models with Zero-init Attention](https://github.com/ZrrSkywalker/LLaMA-Adapter)
```

**Review by pacman100 — APPROVED** [review]
Thank you @MKhalusova for updating the doc page with the supported models tables, LGTM! 🚀 

Left a suggestion based on recently merged PRs

**Pull request merged**